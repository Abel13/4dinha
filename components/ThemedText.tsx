import { Text, type TextProps, StyleSheet } from 'react-native';

import { useThemeColor } from '@/hooks/useThemeColor';
import { Colors } from '@/constants/Colors';

export type ThemedTextProps = TextProps & {
  darkColor?: string;
  type?:
    | 'default'
    | 'title'
    | 'defaultSemiBold'
    | 'subtitle'
    | 'h4'
    | 'link'
    | 'error';
};

export const ThemedText = ({
  style,
  darkColor,
  type = 'default',
  ...rest
}: ThemedTextProps) => {
  const color = useThemeColor({ dark: darkColor }, 'text');

  return (
    <Text
      style={[
        { color },
        type === 'default' ? styles.default : undefined,
        type === 'title' ? styles.title : undefined,
        type === 'defaultSemiBold' ? styles.defaultSemiBold : undefined,
        type === 'subtitle' ? styles.subtitle : undefined,
        type === 'h4' ? styles.h4 : undefined,
        type === 'link' ? styles.link : undefined,
        type === 'error' ? styles.error : undefined,
        style,
      ]}
      {...rest}
    />
  );
};

const styles = StyleSheet.create({
  default: {
    fontSize: 16,
    lineHeight: 24,
    fontFamily: 'BarlowCondensed',
  },
  defaultSemiBold: {
    fontSize: 16,
    lineHeight: 24,
    fontWeight: '600',
    fontFamily: 'BarlowCondensedSemiBold',
  },
  title: {
    fontSize: 32,
    fontWeight: 'bold',
    lineHeight: 32,
    fontFamily: 'BarlowCondensedBold',
  },
  subtitle: {
    fontSize: 20,
    fontWeight: 'bold',
    fontFamily: 'BarlowCondensedBold',
  },
  h4: {
    fontSize: 16,
    fontWeight: 'bold',
    fontFamily: 'BarlowCondensedBold',
  },
  link: {
    lineHeight: 30,
    fontSize: 16,
    color: Colors.dark.link,
  },
  error: {
    lineHeight: 16,
    fontSize: 12,
    color: Colors.dark.danger,
  },
});
