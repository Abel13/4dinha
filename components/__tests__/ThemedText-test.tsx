import * as React from 'react';
import { render } from '@testing-library/react-native';

import { ThemedText } from '../ThemedText';

describe('<ThemedText />', () => {
  test('CustomText renders correctly', () => {
    const tree = render(<ThemedText>Some text</ThemedText>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Text renders correctly on HomeScreen', () => {
    const { getByText } = render(<ThemedText>Welcome!</ThemedText>);

    const text = getByText('Welcome!');
    expect(text).toBeTruthy();
  });
});
