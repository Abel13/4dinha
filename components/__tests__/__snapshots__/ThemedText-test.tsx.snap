// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThemedText /> CustomText renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#151718",
      },
      {
        "fontFamily": "BarlowCondensed",
        "fontSize": 16,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  Some text
</Text>
`;
